nifi-nar-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/PublishSlack.java

```diff
@@ -0,0 +1,517 @@
+            getLogger().error("Failed to resolve Slack Channel ID for {}; transferring to {}", flowFile, relationship, e);
+            return null;
+        }
+    }
+
+    private void publishContentAsMessage(FlowFile flowFile, final String channelId, final ProcessContext context, final ProcessSession session) {
+        // Slack limits the message size to 100,000 characters. We don't have a way to know based on the size of the FlowFile how many characters it will contain,
+        // but we can be rather certain that if the size exceeds 500,000 bytes, it will also exceed 100,000 characters. As a result, we pre-emptively route to
+        // 'too large' in order to avoid buffering the contents into memory.
+        if (flowFile.getSize() > 500_000) {
+            getLogger().error("Cannot send contents of FlowFile {} to Slack because its length exceeds 500,000 bytes; routing to 'failure'");
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        final String charsetName = context.getProperty(CHARACTER_SET).evaluateAttributeExpressions(flowFile).getValue();
+
+        final byte[] buffer = new byte[(int) flowFile.getSize()];
+        final String messageText;
+        try (final InputStream in = session.read(flowFile)) {
+            StreamUtils.fillBuffer(in, buffer, true);
+            messageText = new String(buffer, charsetName);
+        } catch (final IOException ioe) {
+            getLogger().error("Failed to send contents of FlowFile {} to Slack; routing to failure", ioe);
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        if (messageText.length() > 100_000) {
+            getLogger().error("Cannot send contents of FlowFile {} to Slack because its length exceeds 100,000 characters; routing to 'failure'");
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        publishAsMessage(flowFile, channelId, messageText, context, session);
+    }
+
+    private void publishAsMessage(FlowFile flowFile, final String channelId, final String messageText, final ProcessContext context, final ProcessSession session) {
+        final String threadTs = context.getProperty(THREAD_TS).evaluateAttributeExpressions(flowFile).getValue();
+
+        final ChatPostMessageRequest request = ChatPostMessageRequest.builder()
+            .channel(channelId)
+            .text(messageText)
+            .threadTs(threadTs)
+            .build();
+
+        final ChatPostMessageResponse postMessageResponse;
+        try {
+            postMessageResponse = client.chatPostMessage(request);
+        } catch (final Exception e) {
+            final Relationship relationship = handleClientException(channelId, flowFile, session, context, e);
+            getLogger().error("Failed to send message to Slack for {}; transferring to {}", flowFile, e, relationship);
```

Comment:
It looks like the exception and relationship are reversed:
```suggestion
            getLogger().error("Failed to send message to Slack for {}; transferring to {}", flowFile, relationship, e);
```